Model/Api/ShippingMethods.php

```diff
@@ -310,27 +313,27 @@ public function getShippingMethods($cart, $shipping_address)
             list(, $quoteId) = explode(' / ', $cart['display_id']);
 
             // Load quote from entity id
-            $quote = $this->getQuoteById($quoteId);
+            $this->quote = $this->getQuoteById($quoteId);
 
-            if (!$quote) {
+            if (!$this->quote) {
                 $this->throwUnknownQuoteIdException($quoteId);
             }
 
-            $this->preprocessHook($quote->getStoreId());
+            $this->preprocessHook($this->quote->getStoreId());
 
-            $this->checkCartItems($cart, $quote);
+            $this->checkCartItems($cart, $this->quote);
```

Comment:
Does it make sense to avoid here using `$this->quote`?
Can we just call `$this->quote` directly in protected method instead of transfer it through parameter?

The same for `$this->preprocessHook($this->quote->getStoreId());`

Reply:
Makes sense